src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs

```diff
@@ -670,6 +670,11 @@ public ValueTask SendFileAsync(string? fileName, CancellationToken cancellationT
         /// <exception cref="SocketException">An error occurred when attempting to access the socket.</exception>
         public ValueTask SendFileAsync(string? fileName, ReadOnlyMemory<byte> preBuffer, ReadOnlyMemory<byte> postBuffer, TransmitFileOptions flags, CancellationToken cancellationToken = default)
         {
+            if (_protocolType == ProtocolType.Udp)
+            {
+                throw new NotSupportedException(SR.net_notconnected);
+            }
```

Comment:
Doesn't the check `if (!IsConnectionOriented)` do the job? In #47472 I only tested the sync overload.